src/combinatorics/simple_graph/basic.lean

```diff
@@ -747,66 +747,53 @@ end incidence
 
 /-! ## Edge deletion -/
 
+section delete_edges
+variables {s s₁ s₂ : set (sym2 V)}
+
 /-- Given a set of vertex pairs, remove all of the corresponding edges from the
 graph's edge set, if present.
 
 See also: `simple_graph.subgraph.delete_edges`. -/
-def delete_edges (s : set (sym2 V)) : simple_graph V :=
-{ adj := G.adj \ sym2.to_rel s,
-  symm := λ a b, by simp [adj_comm, sym2.eq_swap] }
+def delete_edges (s : set (sym2 V)) : simple_graph V := G \ from_edge_set s
 
-@[simp] lemma delete_edges_adj (s : set (sym2 V)) (v w : V) :
-  (G.delete_edges s).adj v w ↔ G.adj v w ∧ ¬ ⟦(v, w)⟧ ∈ s := iff.rfl
+@[simp] lemma delete_edges_adj :
+  (G.delete_edges s).adj v w ↔ G.adj v w ∧ ¬ ⟦(v, w)⟧ ∈ s :=
+and_congr_right $ λ h, iff.not $ and_iff_left h.ne
 
-lemma sdiff_eq_delete_edges (G G' : simple_graph V) :
-  G \ G' = G.delete_edges G'.edge_set :=
-by { ext, simp }
-
-lemma delete_edges_eq_sdiff_from_edge_set (s : set (sym2 V)) :
```

Comment:
Rather than delete this, how about leave a definition lemma?

Reply:
Isn't the equational lemma good enough for you?